### PR TITLE
Fish spec

### PR DESCRIPTION
**Pull request by bpinto**
@bucaran and I created this plugin called [fish-spec](https://github.com/bpinto/oh-my-fish/tree/master/plugins/fish-spec) to test [oh-my-fish](https://github.com/bpinto/oh-my-fish) internals and plugins.

This is fish-spec running tests on verbose mode:
b547f8ac-9e33-11e4-8825-35c97e9429dd

The aim of this pull request is to discuss the introduction of fish-spec as a dependency to fish-shell tests so that we can start writing tests in an easier way to understand than we currently do (test1.in, test1.out).

P.S.: The only interesting commit to be seen is the [last one](https://github.com/fish-shell/fish-shell/commit/894d75cebba75b6ea6d4b8008c9478c2d7267342).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ FROM centos:latest
 
 # Build dependency
 RUN yum update -y &&\
-  yum install -y autoconf automake clang gcc-c++ make ncurses-devel &&\
+  yum install -y autoconf automake bc clang gcc-c++ make ncurses-devel &&\
```
Comment (bpinto) [inline Dockerfile]: This will be added in a separate PR to fish as `math` requires `bc` package.

**Comment by bpinto** [comment]
This is the output of the added tests, with and without verbose mode.

image

**Comment by bucaran** [comment]
Sometimes mistakes happen, here is how the output looks like then.

533f50d4-9e05-11e4-86a1-6f1d3d10c8bf

**Comment by krader1961** [comment]
@bpinto and @bucaran: This pull-request is now more than a year old and the https://github.com/bpinto/oh-my-fish/tree/master/plugins/fish-spec link no longer works. Is this still something you'd like to see merged? If yes would you mind opening an enhancement issue that describes the problem and your solution?

**Comment by bucaran** [comment]
Not me. I am also not working on `fish-spec` anymore and writing unit tests using [Fishtape](https://github.com/fishery/fishtape).

**Comment by krader1961** [comment]
I'd love to see a framework for fish unit tests in the same vein as Python's unittest2. There are some intriguing ideas in this change but also some problems (like the "it_supports_" naming scheme and redefining `expect` as a function). Given that no one else has advocated for adopting this in over a year I think it's time to close this and go back to the drawing board.

**Comment by bucaran** [comment]
@krader1961 Have you had a look at Fishtape? I certainly took inspiration from unittest2, but decided that if it didn't produce a TAP output I might as well give up :smile: 

Tests look like this:

```fish
test "math still works"
    5 -eq (math 2 + 3)
end

test "grep still finds"
    dog = (printf "%s\n" $animals | grep dog)
end

test "file exists"
    -f $my_file
end 
```
